features/recipes/models.py

```diff
@@ -1,10 +1,70 @@
+import enum
+
 from features import DbBaseModel
 from sqlalchemy.orm import Mapped, mapped_column, relationship
 from sqlalchemy import String, Integer, Float, func, ForeignKey, DateTime, Boolean
 import datetime
 from typing import Optional
 
 
+class IngredientCategory(DbBaseModel):
+
+    """Ingredient category model"""
+
+    __tablename__ = "INGREDIENT_CATEGORIES"
+
+    id: Mapped[int] = mapped_column(
+        Integer, autoincrement=True, primary_key=True, init=False
+    )
+    name: Mapped[str] = mapped_column(String(255), unique=True)
+    created_by: Mapped[int] = mapped_column(ForeignKey("Users.id"))
+    created_on: Mapped[datetime.datetime] = mapped_column(
+        DateTime, server_default=func.current_timestamp(), init=False
+    )
+
+    updated_by: Mapped[Optional[int]] = mapped_column(
+        ForeignKey("Users.id"), nullable=True, init=False
+    )
+
+    updated_on: Mapped[datetime.datetime] = mapped_column(
+        DateTime,
+        server_default=func.current_timestamp(),
+        onupdate=func.current_timestamp(),
+        init=False,
+    )
+
+    ingredients: Mapped[list["Ingredient"]] = relationship(
+        "Ingredient", back_populates="ingredient", init=False, lazy="selectin"
+    )
+
+
+class MeasurementUnits(enum.Enum):
```

Comment:
This is not need here anymore. If you need it input model for validations, move it there